translationRecorder/app/src/main/java/org/wycliffeassociates/translationrecorder/ProjectManager/activities/ActivityUnitList.java

```diff
@@ -187,18 +189,29 @@ public void onNegativeClick(RatingDialog dialog) {
 
 
     private void prepareUnitCardData() {
-        try {
-            ChunkPlugin chunkPlugin = mProject.getChunkPlugin(new ChunkPluginLoader(this));
+        if(chunkPlugin != null) {
             List<Chunk> chunks = chunkPlugin.getChapter(mChapterNum).getChunks();
             for (Chunk unit : chunks) {
                 String title = Utils.capitalizeFirstLetter(mProject.getModeName()) + " " + unit.getLabel();
-                mUnitCardList.add(new UnitCard(mAdapter, mProject, title, mChapterNum, unit.getStartVerse(), unit.getEndVerse()));
+                mUnitCardList.add(new UnitCard(
+                        mAdapter,
+                        mProject,
+                        title,
+                        mChapterNum,
+                        unit.getStartVerse(),
+                        unit.getEndVerse(),
+                        this
+                ));
             }
-        } catch (IOException e) {
-            e.printStackTrace();
         }
     }
 
+    @Override
+    public void onTakeDeleted() {
+        ProjectProgress pp = new ProjectProgress(mProject, db, chunkPlugin.getChapters());
```

Comment:
this can be moved to the top; I'd name it projectProgress over pp though since its use will be further from the declaration. (you're in the unit list, so everything is in the same project, this will cut down on a lot of work being done repeatedly)